src/main/java/com/tsmms/skoop/notification/NotificationRepository.java

```diff
@@ -47,7 +47,9 @@
 			" OPTIONAL MATCH (n)-[r6:SKILL]->(skill:Skill) " +
 			" WITH n, r1, registration, r2, registeredUser, r3, c, r4, community, r5, user, r6, skill " +
 			" OPTIONAL MATCH (n)-[r7:USER_PROJECT]->(up:UserProject)-[r8:USER]->(u:User) " +
-			" RETURN n, r1, registration, r2, registeredUser, r3, c, r4, community, r5, user, r6, skill, r7, up, r8, u" +
+			" WITH n, r1, registration, r2, registeredUser, r3, c, r4, community, r5, user, r6, skill, r7, up, r8, u " +
+			" OPTIONAL MATCH (n)-[r7:USER_PROJECT]->(up:UserProject)-[r9:PROJECT]->(project:Project) " +
```

Comment:
`(n)-[r7:USER_PROJECT]->(up:UserProject)` can be replaced with `up` since a user project was fetched at the previous stage. To have a better performance I suggest to use the previously fetched `up` reference.